src/txn/txn_timestamp.c

```diff
@@ -653,6 +656,13 @@ __txn_set_commit_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t commit_ts)
     if (!F_ISSET(txn, WT_TXN_HAS_TS_DURABLE))
         txn->durable_timestamp = commit_ts;
 
+    /* Don't be overly greedy about updating the commit timestamp, it's shared */
+    WT_ACQUIRE_READ_WITH_BARRIER(newest_commit_ts, txn_global->newest_seen_timestamp);
```

Comment:
We should use WT_ACQUIRE_READ. WT_ACQUIRE_READ_WITH_BARRIER is about to deprecate.